test/components/views/location/LocationPicker-test.tsx

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2021 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import sdk from "../../../skinned-sdk";
+import { getGeoUri } from "../../../../src/components/views/location/LocationPicker";
+
+sdk.getComponent("LocationPicker");
```

Comment:
what is this doing?

Reply:
Without it I get
```
$ jest LocationButton
 FAIL  test/components/views/location/LocationButton-test.tsx
  ● Test suite failed to run

    Attempted to get a component (views.elements.DialogButtons) before a skin has been loaded. This is probably because either: a) Your app has not called sdk.loadSkin(), or b) A component has called getComponent at the root level

      31 |         if (!name) throw new Error(`Invalid component name: ${name}`);
      32 |         if (this.components === null) {
    > 33 |             throw new Error(
         |                   ^
      34 |                 `Attempted to get a component (${name}) before a skin has been loaded.`+
      35 |                 " This is probably because either:"+
      36 |                 " a) Your app has not called sdk.loadSkin(), or"+

      at Skinner.getComponent (src/Skinner.ts:33:19)
      at Object.getComponent (src/index.ts:33:20)
      at src/utils/replaceableComponent.ts:42:22
      at Object.<anonymous> (src/components/views/elements/DialogButtons.tsx:24:89)
      at Object.<anonymous> (src/components/views/location/LocationPicker.tsx:23:1)
      at Object.<anonymous> (src/components/views/location/LocationButton.tsx:23:1)
      at Object.<anonymous> (test/components/views/location/LocationButton-test.tsx:18:1)

Test Suites: 1 failed, 1 total
Tests:       0 total
Snapshots:   0 total
Time:        1.552 s, estimated 4 s
Ran all test suites matching /LocationButton/i.
(node:311191) UnhandledPromiseRejectionWarning: TypeError: Cannot read property '_origin' of null
    at Window.get localStorage [as localStorage] (/home/andy/code/public/element-web/matrix-react-sdk/node_modules/jsdom/lib/jsdom/browser/Window.js:420:50)
    at DeviceSettingsHandler.isSupported (/home/andy/code/public/element-web/matrix-react-sdk/src/settings/handlers/DeviceSettingsHandler.ts:152:5)
    at Function.isLevelSupported (/home/andy/code/public/element-web/matrix-react-sdk/src/settings/SettingsStore.ts:502:34)
    at Function.getHandlers (/home/andy/code/public/element-web/matrix-react-sdk/src/settings/SettingsStore.ts:659:25)
    at Function.getHandler (/home/andy/code/public/element-web/matrix-react-sdk/src/settings/SettingsStore.ts:648:36)
    at Function.setValue (/home/andy/code/public/element-web/matrix-react-sdk/src/settings/SettingsStore.ts:434:35)
    at /home/andy/code/public/element-web/matrix-react-sdk/src/languageHandler.tsx:379:28
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:311191) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
(node:311191) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Reply:
`import sdk from "../../../skinned-sdk";` should fix that - `sdk.getComponent("LocationPicker");` should just be looking up a dict and returning a value

Reply:
Ah thanks, `import "../../../skinned-sdk";` was the recipe I needed to avoid a warning about not using `sdk`

Reply:
Most of the tests files say this at the top of them

`import '../skinned-sdk'; // Must be first for skinning to work`